### PR TITLE
Update references to `python-build-standalone` to reflect the transferred project

### PR DESCRIPTION
**Pull request by zanieb**

### CODE REVIEW

**Review by charliermarsh — APPROVED** [review]

File: crates/uv-python/src/downloads.rs [hunk 1]
```diff
@@ -658,7 +658,9 @@ impl ManagedPythonDownload {
                 if let Some(mirror) = python_install_mirror {
                     let Some(suffix) = self.url.strip_prefix(
                         "https://github.com/indygreg/python-build-standalone/releases/download/",
```
Comment (charliermarsh) [inline crates/uv-python/src/downloads.rs:660]: Is this necessary to retain?

Comment (zanieb) [reply crates/uv-python/src/downloads.rs:660]: I think it'd be a breaking change otherwise, yes.

Comment (zanieb) [reply crates/uv-python/src/downloads.rs:660]: Oh nevermind we strip _our_ prefix. Yeah I guess we can drop it?

Comment (zanieb) [reply crates/uv-python/src/downloads.rs:660]: Why do we throw an error here?

```
                    else {
                        return Err(Error::Mirror(EnvVars::UV_PYTHON_INSTALL_MIRROR, self.url));
                    };
```

This is what confused me. Our own URL doesn't meet our expectations?

Comment (charliermarsh) [reply crates/uv-python/src/downloads.rs]: Yeah

**Pull request merged**